src/utils/vcs.rs

```diff
@@ -922,6 +929,64 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_extract_provider_name() {
+        // Test basic provider name extraction
+        assert_eq!(extract_provider_name("github.com"), "github");
+        assert_eq!(extract_provider_name("gitlab.com"), "gitlab");
+        assert_eq!(extract_provider_name("bitbucket.org"), "bitbucket");
+
+        // Test edge case with trailing dots
+        assert_eq!(extract_provider_name("github.com."), "github");
+        assert_eq!(extract_provider_name("gitlab.com.."), "gitlab");
```

Comment:
Having two consecutive dots in a URL is invalid, so we don't need to test this case imo (behavior can be left undefined)

```suggestion
```